Source/Core/Duality/Launcher/DualityLauncher.cs

```diff
@@ -41,23 +43,23 @@ public static void Run(string[] args)
 			// Write initial log message before actually booting Duality
 			Logs.Core.Write("Running DualityLauncher with flags: {1}{0}",
 				Environment.NewLine,
-				new[] { isDebugging ? "Debugging" : null, isProfiling ? "Profiling" : null, isRunFromEditor ? "RunFromEditor" : null }.NotNull().ToString(", "));
+				this.launcherArgs);
 
 			// Initialize the Duality core
 			DualityApp.Init(
 				DualityApp.ExecutionEnvironment.Launcher,
 				DualityApp.ExecutionContext.Game,
 				new DefaultAssemblyLoader(),
-				args);
+				this.launcherArgs);
 
 			// Open up a new window
 			WindowOptions options = new WindowOptions
 			{
 				Size = DualityApp.UserData.WindowSize,
-				ScreenMode = isDebugging ? ScreenMode.Window : DualityApp.UserData.WindowMode,
-				RefreshMode = (isDebugging || isProfiling) ? RefreshMode.NoSync : DualityApp.UserData.WindowRefreshMode,
+				ScreenMode = this.launcherArgs.IsDebugging ? ScreenMode.Window : DualityApp.UserData.WindowMode,
+				RefreshMode = (this.launcherArgs.IsDebugging || this.launcherArgs.IsProfiling) ? RefreshMode.NoSync : DualityApp.UserData.WindowRefreshMode,
```

Comment:
This is unrelated to your changes, but since you're at it anyway, can you remove the effect `IsDebugging` has on `RefreshMode`? 

I'm no longer sure why I added this way back, but I think it's a bad idea to always debug with infinite FPS when the game actually runs _with_ VSync in production. Stuff just behaves slightly different at 60 FPS vs 500 FPS with variable frame time and `TimeMult` / `DeltaTime` isn't designed to deal with differences _that_ large. Better to let people optimize for / test in the case that players will actually experience.

Reply:
How about the ScreenMode as that one also changes when debugging?